libcylinder/src/signature.rs

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2017 Intel Corporation
+ * Copyright 2018-2020 Cargill Incorporated
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ------------------------------------------------------------------------------
+ */
+
+use crate::error::SignatureParseError;
+use crate::hex;
+
+#[derive(PartialEq, Eq)]
+pub struct Signature(Vec<u8>);
+
+/// A signature
+impl Signature {
+    /// Constructs a new Signature from a given set of bytes.
+    pub fn new(sig_bytes: Vec<u8>) -> Self {
+        Self(sig_bytes)
+    }
+
+    /// Creates a new Signature from a hex string.
+    ///
+    /// # Errors
+    ///
+    /// Returns a SignatureParseError if the provided signature string is not valid hex.
+    pub fn from_hex(sig_hex: &str) -> Result<Self, SignatureParseError> {
+        hex::hex_str_to_bytes(sig_hex)
+            .map(Self)
+            .map_err(|e| SignatureParseError(e.to_string()))
+    }
+
+    /// Creates a new Signature from a hex string, panicking if not valid hex.
```

Comment:
Can we make a `# Panics` section for the panic info? IMO we should make it as obvious as possible when anything panics.

Reply:
Dropped this function